bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandWebSockets.java

```diff
@@ -0,0 +1,203 @@
+
+        public TouchWandSocket() {
+        }
+
+        @OnWebSocketClose
+        public void onClose(int statusCode, String reason) {
+            logger.debug("Connection closed: {} - {}", statusCode, reason);
+            if (!isShutDown) {
+                logger.debug("weSocket Closed - reconnecting");
+                asyncWeb();
+            } else {
+                this.session = null;
+            }
+        }
+
+        @OnWebSocketConnect
+        public void onConnect(Session session) {
+            logger.debug("TouchWandWebSockets connected to {}", session.getRemoteAddress().toString());
+            this.session = session;
+            try {
+                session.getRemote().sendString("{\"myopenhab\": \"myopenhab\"}");
+            } catch (IOException e) {
+                logger.warn("sendString : {}", e.getMessage());
+            }
+        }
+
+        @OnWebSocketMessage
+        public void onMessage(String msg) {
+            TouchWandUnitData touchWandUnit;
+            JsonParser jsonParser = new JsonParser();
+            try {
+                JsonObject unitObj = jsonParser.parse(msg).getAsJsonObject();
+                boolean eventUnitChanged = unitObj.get("type").getAsString().equals("UNIT_CHANGED");
+                if (!eventUnitChanged) {
+                    return;
+                }
+                touchWandUnit = TouchWandUnitFromJson.parseResponse(unitObj.get("unit").getAsJsonObject());
+                if (!touchWandUnit.getStatus().equals("ALIVE")) {
+                    return;
+                }
+                boolean supportedUnitType = Arrays.asList(SUPPORTED_TOUCHWAND_TYPES).contains(touchWandUnit.getType());
+                if (!supportedUnitType) {
+                    logger.debug("UNIT_CHANGED for unsupported unit type {}", touchWandUnit.getType());
+                    return;
+                }
+                logger.debug("UNIT_CHANGED: name {} id {} status {}", touchWandUnit.getName(), touchWandUnit.getId(),
+                        touchWandUnit.getCurrStatus());
+                if (touchWandUnit.getCurrStatus() != null) {
+                    for (TouchWandUnitStatusUpdateListener listener : listeners) {
+                        listener.onDataReceived(touchWandUnit);
+                    }
+                }
+            } catch (JsonSyntaxException e) {
+                logger.warn("jsonParser.parse {} ", e.getMessage());
+            }
+        }
+
+        @OnWebSocketError
+        public void onError(Throwable cause) {
+            logger.warn("WebSocket Error: {}", cause.getMessage());
+            if (!isShutDown) {
+                logger.debug("WebSocket onError - reconnecting");
+                asyncWeb();
+            } else {
+                this.session = null;
+            }
+        }
+
+        @SuppressWarnings("null")
+        private void asyncWeb() {
+            if (socketReconnect == null || socketReconnect.isDone()) {
```

Comment:
The null suppression can be removed if you store `socketReconnect` to a local variable and do the null check on that.

Reply:
I am using a global variable and I am trying to prevent a case where there are more then one websocket open .

Reply:
As you are using a global variable, you should store the reference into a local variable and do the nullcheck and the further processing in this method on that, because the global variable could be set to null between `socketReconnect == null` and `socketReconnect.isDone()`.

Reply:
Made the change , hopefully correctly